collections.js

```diff
@@ -7,7 +7,7 @@ global.List = require("./list");
 global.Set = require("./set");
 global.Map = require("./map");
 global.MultiMap = require("./multi-map");
-global.WeakMap = require("weak-map");
+global.WeakMap = window.WeakMap || require("weak-map");
```

Comment:
The WeakMap module handles the fallback case and exports the global if it exists.